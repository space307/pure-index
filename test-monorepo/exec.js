import cp from 'node:child_process'
import path from 'node:path'
import { fileURLToPath } from 'node:url'
import { promisify } from 'node:util'

const filename = importMeta =>
  importMeta.url ? fileURLToPath(importMeta.url) : ''

const dirname = importMeta => path.dirname(filename(importMeta))

export const exec = params => {
  const prefix = path.join(dirname(import.meta), params.path)

  return promisify(cp.exec)(`${params.cmd} --prefix ${prefix}`)
}
