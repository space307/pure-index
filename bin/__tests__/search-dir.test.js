import { join } from 'node:path'
import { expect, test } from 'vitest'

import { exec } from './monorepo/exec.js'

test('package-d', async () => {
  try {
    const { stdout, stderr } = await exec({
      cmd: 'npm run check-exports',
      path: '../package-d'
    })
    expect(stdout).toMatchSnapshot()
    expect(stderr).toMatchSnapshot()
  } catch (e) {
    expect(e.stdout).toMatchSnapshot()
  }
})

test('package-d unhappy', async () => {
  try {
    const { stdout, stderr } = await exec({
      cmd: 'npm run check-exports-unhappy',
      path: '../package-d'
    })
    expect(stdout).toMatchSnapshot()
    expect(stderr).toMatchSnapshot()
  } catch (e) {
    expect(e.stdout).toMatchSnapshot()
  }
})

test('package-d happy', async () => {
  try {
    const { stdout, stderr } = await exec({
      cmd: 'npm run check-exports-happy',
      path: '../package-d'
    })
    expect(stdout).toMatchSnapshot()
    expect(stderr).toMatchSnapshot()
  } catch (e) {
    expect(e.stdout).toMatchSnapshot()
  }
})
