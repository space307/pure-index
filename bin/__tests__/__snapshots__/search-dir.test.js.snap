// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`package-d 1`] = `
"
> check-exports
> node ../../index.js -s ../monorepo/app


 Failed  Unused exports in package-d package found

[
  "ususedFn"
] 

"
`;

exports[`package-d happy 1`] = `
"
> check-exports-happy
> node ../../index.js -s ../monorepo

"
`;

exports[`package-d happy 2`] = `
"âœ” Checking exports from the package-d package
"
`;

exports[`package-d unhappy 1`] = `
"
> check-exports-unhappy
> node ../../index.js -s ../biba


 Failed  Nothing is imported from package-d. Remove it.

"
`;
